src/collections/_documentation/performance/distributed-tracing.md

```diff
@@ -12,105 +12,325 @@ Sentry's Performance features are currently in beta. For more details about acce
 
-View transaction events by clicking on the "Transactions" pre-built query in [Discover]({%- link _documentation/performance/discover/index.md -%}), or by using a search condition `event.type:transaction` in a [Discover Query Builder]({%- link _documentation/performance/discover/query-builder.md -%}) view.
+**Nested Spans**
 
-When you open a transaction event in Discover, you'll see the **span view** at the top of the page. Other information the SDK captured as part of the Transaction event, such as breadcrumbs, will also be displayed.
+Though our examples above had four levels in their hierarchy (trace, transaction, span, child span) there's no set limit to how deep the nesting of spans can go. There are, however, practical limits: transaction payloads sent to Sentry have a maximum allowed size, and as with any kind of logging, there's a balance to be struck between your data's granularity and its usability. 
 
-[{% asset performance/discover-span.png alt="Discover span showing the map of the transactions (aka minimap) and the black dashed handlebars for adjusting the window selection." %}]({% asset performance/discover-span.png @path %})
+**Zero-duration Spans**
+
+It's possible for a span to have equal start and end times, and therefore be recorded as taking no time. This can occur either because the span is being used as a marker (such as is done in [the browser's Performance API](https://developer.mozilla.org/en-US/docs/Web/API/Performance/mark)) or because the amount of time the operation took is less than the measurement resolution (which will vary by service).
+
+**Clock Skew**
+
+If you are collecting transactions from multiple machines, you will likely encounter **clock skew**, where timestamps in one transaction don't align with timestamps in another. For example, if your backend makes a database call, the backend transaction logically should start before the database transaction does. But if the system time on each machine (those hosting your backend and database, respectively) isn't synced to common standard, it's possible that won't be the case. It's also possible for the ordering to be correct, but for the two recorded timeframes to not line up in a way that accurately reflects what actually happened.
+
+Because there is no way for Sentry to judge either the relative or absolute accuracy of your timestamps, it does not attempt to correct for this. And while you can reduce clock skew by using Network Time Protocol (NTP) or your cloud provider's clock synchronization services, you may still notice small discrepancies in your data, as synchronizing clocks on small intervals is challenging.
```

Comment:
@HazAT and I talked about this, and our conclusion was that

1) there's a difference between clock _skew_ (spans being out of sync with each other) and clock _drift_, (the spans being in sync with each other, but somehow wildly off from our clock, such that a new transaction reads as having happened weeks ago) which is what I believe you're talking about,

2) relay doesn't correct skew but it does correct drift, but

3) the former was worth documenting but the latter was not, because skew is much more likely and much more noticeable, whereas drift severe enough to make us change it was going to be very rare.

Does any of that go against what you know and/or, do you think we should document drift as well as skew?

Reply:
Long story short, we can avoid all this by not making claims about what Sentry does or does not, as that's an implementation detail that can easily change.

```suggestion
We suggest you use the Network Time Protocol (NTP) or your cloud provider's clock synchronization services to reduce clock skew.
```

<details>
<summary>Details</summary>

1. https://en.wikipedia.org/wiki/Clock_drift & https://en.wikipedia.org/wiki/Clock_skew
2. The source code links I shared use the name "drift", but I don't think it matches the definition of clock drift. It corrects for the **skew** between the client clock and the clock of the Relay server (however somehow ignoring the data transfer and processing time). In practice, for JS transactions at the moment we never send the `sent_at` value, thus Relay does:
    ```
    event.start_timestamp += current_time_at_normalization_time - event.timestamp
    ```
    and
    ```
    event.timestamp += current_time_at_normalization_time - event.timestamp
    ```

    This is equivalent to:
    ```
    event.timestamp = current_time_at_normalization_time
    event.start_timestamp = event.timestamp - duration
    ```

    (Unless I'm reading the Relay code wrong!)
3. --

My comment was re:

> Because there is no way for Sentry to judge either the relative or absolute accuracy of your timestamps, it does not attempt to correct for this.

In fact, we DO judge the RELATIVE accuracy of the timestamps AND DO attempt to correct the timestamps. We always do it, for every ingested transaction.

---

**Food for thought**

I've discussed the Relay implementation with Markus before... if your multiple servers have their clocks well synced and produce trustworthy timestamps, but all have quite different latencies sending data to Relay (or even more simply -- Relay has a non-constant processing time), then the "correction" in Relay may really be mixing up the relative timeframes of transactions in a trace.

https://github.com/getsentry/relay/issues/534

</details>

Reply:
Ah, all good to know, thank you! And you do raise an interesting food-for-thought question...

But yes, long story short, I think your suggestion to just not mention it is a good one.